scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -825,7 +825,10 @@ class FossId internal constructor(
                 val snippetProvenance = it.url?.let { url ->
                     ArtifactProvenance(RemoteArtifact(url, Hash.NONE))
                 } ?: UnknownProvenance
-                val purlType = it.url?.let { url -> urlToPackageType(url).toString() }
+                val purl = it.purl ?: run {
+                    val purlType = it.url?.let { url -> urlToPackageType(url).toString() }
```

Comment:
Hmm, this should actually conflict with your other (recently merged) PR and should be rebased.